Libraries/IonAuth.php

```diff
@@ -0,0 +1,440 @@
+<?php namespace IonAuth\Libraries;
+
+/**
+ * Name:    Ion Auth
+ * Author:  Ben Edmunds
+ *           ben.edmunds@gmail.com
+ *           @benedmunds
+ *
+ * Added Awesomeness: Phil Sturgeon
+ *
+ * Created:  10.01.2009
+ *
+ * Description:  Modified auth system based on redux_auth with extensive customization. This is basically what Redux Auth 2 should be.
+ * Original Author name has been kept but that does not mean that the method has not been modified.
+ *
+ * Requirements: PHP7.1 or above
+ *
+ * @package    CodeIgniter-Ion-Auth
+ * @author     Ben Edmunds
+ * @license    https://opensource.org/licenses/MIT	MIT License
+ * @link       http://github.com/benedmunds/CodeIgniter-Ion-Auth
+ * @filesource
+ */
+
+/**
+ * Class IonAuth
+ */
+class IonAuth
+{
+	/**
+	 * @var \IonAuth\Config\IonAuth
+	 */
+	private $config;
+
+	/**
+	 *
+	 * @var \IonAuth\Models\IonAuthModel
+	 */
+	private $ionAuthModel;
+
+	/**
+	 * Extra where
+	 *
+	 * @var array
+	 **/
+	public $_extra_where = [];
+
+	/**
+	 * Caching of users and their groups
+	 *
+	 * @var array
+	 **/
+	public $_cacheUserInGroup;
+
+	/**
+	 * @var \CodeIgniter\Email\Email
+	 */
+	protected $email;
+
+	/**
+	 * __construct
+	 *
+	 * @author Ben
+	 */
+	public function __construct()
+	{
+		// Check compat first
+		$this->checkCompatibility();
+
+		$this->config = config('IonAuth\\Config\\IonAuth');
+
+		$this->email = \Config\Services::email();
+		//$this->load->helper(['cookie', 'language','url']);
+
+		//$this->session = \Config\Services::session();   //$this->load->library('session');
```

Comment:
Set this->session here so we can use it elsewhere, should allow easier testing later
```suggestion
		$this->session = session();
```